Windows compiler warnings

FYI: MSVC currently gives the following warnings:

```
warning C4800: 'int' : forcing value to bool 'true' or 'false' (performance warning) (C:\Projects\ogs\GeoLib\PointVec.cpp)  C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\include\utility  144 1   GeoLib

warning C4800: 'int' : forcing value to bool 'true' or 'false' (performance warning) (C:\Projects\ogs\GeoLib\Surface.cpp)   C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\include\utility  144 1   GeoLib

warning C4244: 'initializing' : conversion from '__int64' to 'int', possible loss of data (C:\Projects\ogs\MathLib\LinAlg\Eigen\EigenLinearSolver.cpp)  c:\projects\ogsbuild\external\eigen\src\eigen\eigen\src\SparseLU\SparseLU_SupernodalMatrix.h    236 1   MathLib

warning C4244: 'initializing' : conversion from '__int64' to 'int', possible loss of data (C:\Projects\ogs\MathLib\LinAlg\Eigen\EigenLinearSolver.cpp)  c:\projects\ogsbuild\external\eigen\src\eigen\eigen\src\SparseLU\SparseLU_SupernodalMatrix.h    237 1   MathLib

warning C4244: 'initializing' : conversion from '__int64' to 'int', possible loss of data (C:\Projects\ogs\MathLib\LinAlg\Eigen\EigenLinearSolver.cpp)  c:\projects\ogsbuild\external\eigen\src\eigen\eigen\src\SparseLU\SparseLU.h 733 1   MathLib

warning C4244: 'initializing' : conversion from '__int64' to 'int', possible loss of data (C:\Projects\ogs\MathLib\LinAlg\Eigen\EigenLinearSolver.cpp)  c:\projects\ogsbuild\external\eigen\src\eigen\eigen\src\SparseLU\SparseLU.h 734 1   MathLib

warning C4267: '=' : converSMatrixReorderedOpenMP::amux': virtual function overrides 'MathLib::CRSMatrix<double,unsigned int>::amux', previous versions of the compiler did not override when parameters only differed by const/volatile qualifiers (C:\Projects\ogs\MathLib\LinAlg\Sparse\NestedDissectionPermutation\CRSMatrixReorderedOpenMP.cpp)    C:\Projects\ogs\MathLib\LinAlg\Sparse\NestedDissectionPermutation\CRSMatrixReorderedOpenMP.h    28  1   MathLib
```

Console output is the following, if that helps:

```
14>C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\include\utility(144): warning C4800: 'int' : forcing value to bool 'true' or 'false' (performance warning) (C:\Projects\ogs\GeoLib\Surface.cpp)
14>          c:\projects\ogs\geolib\AABB.h(111) : see reference to function template instantiation 'std::pair<bool,std::bitset<0x03>>::pair<int,int,void>(_Other1 &&,_Other2 &&)' being compiled
14>          with
14>          [
14>              _Other1=int
14>  ,            _Other2=int
14>          ]
14>          c:\projects\ogs\geolib\AABB.h(111) : see reference to function template instantiation 'std::pair<bool,std::bitset<0x03>>::pair<int,int,void>(_Other1 &&,_Other2 &&)' being compiled
14>          with
14>          [
14>              _Other1=int
14>  ,            _Other2=int
14>          ]
14>          C:\Projects\ogs\GeoLib\Surface.cpp(66) : see reference to function template instantiation 'bool GeoLib::AABB::update<GeoLib::Point>(const PNT_TYPE &)' being compiled
14>          with
14>          [
14>              PNT_TYPE=GeoLib::Point
14>          ]
```

Here's an update, the current list of warnings (some are new, some are already listed above):

```
warning C4244: 'argument' : conversion from 'const double' to 'unsigned __int64', possible loss of data	C:\Projects\ogs\NumLib\TimeStepping\Algorithms\FixedTimeStepping.cpp	79	1	NumLib
warning C4244: 'initializing' : conversion from '__int64' to 'int', possible loss of data (C:\Projects\ogs\MathLib\LinAlg\Eigen\EigenLinearSolver.cpp)	c:\projects\ogsbuild\external\eigen\src\eigen\eigen\src\SparseLU\SparseLU.h	733	1	MathLib
warning C4244: 'initializing' : conversion from '__int64' to 'int', possible loss of data (C:\Projects\ogs\MathLib\LinAlg\Eigen\EigenLinearSolver.cpp)	c:\projects\ogsbuild\external\eigen\src\eigen\eigen\src\SparseLU\SparseLU.h	734	1	MathLib
warning C4244: 'initializing' : conversion from '__int64' to 'int', possible loss of data (C:\Projects\ogs\MathLib\LinAlg\Eigen\EigenLinearSolver.cpp)	c:\projects\ogsbuild\external\eigen\src\eigen\eigen\src\SparseLU\SparseLU_SupernodalMatrix.h	236	1	MathLib
warning C4244: 'initializing' : conversion from '__int64' to 'int', possible loss of data (C:\Projects\ogs\MathLib\LinAlg\Eigen\EigenLinearSolver.cpp)	c:\projects\ogsbuild\external\eigen\src\eigen\eigen\src\SparseLU\SparseLU_SupernodalMatrix.h	237	1	MathLib
warning C4800: 'int' : forcing value to bool 'true' or 'false' (performance warning) (C:\Projects\ogs\GeoLib\Surface.cpp)	C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\include\utility	144	1	GeoLib
```

Some more warnings on Windows:
```
warning C4180: qualifier applied to function type has no meaning; ignored (C:\Projects\ogs\Tests\MathLib\TestNonlinear1D.cpp)	C:\Projects\ogs\MathLib\Nonlinear\Root1D.h	52
warning C4180: qualifier applied to function type has no meaning; ignored (C:\Projects\ogs\Tests\MathLib\TestNonlinear1D.cpp)	C:\Projects\ogs\MathLib\Nonlinear\Root1D.h	112
warning C4244: 'argument' : conversion from 'const double' to 'const size_t', possible loss of data	C:\Projects\ogs\Tests\NumLib\TestExtrapolation.cpp	327
warning C4244: 'initializing' : conversion from 'double' to 'int', possible loss of data	C:\Projects\ogs\Tests\GeoLib\TestAABB.cpp	174
warning C4244: 'initializing' : conversion from 'double' to 'int', possible loss of data	C:\Projects\ogs\Tests\GeoLib\TestAABB.cpp	175
warning C4244: 'initializing' : conversion from 'double' to 'int', possible loss of data	C:\Projects\ogs\Tests\GeoLib\TestAABB.cpp	176
```

The current list of windows compiler warnings. I didn't compare to the one's above so pls take this as the new status:

```
warning C4244: '=' : conversion from 'std::streamoff' to 'int', possible loss of data	C:\Projects\ogs\MeshLib\IO\FEFLOW\FEFLOWMeshInterface.cpp	318	MeshLib
warning C4244: 'argument' : conversion from 'const __int64' to 'const unsigned int', possible loss of data (C:\Projects\ogs\ProcessLib\TES\TESLocalAssemblerInner.cpp)	c:\projects\ogs\processlib\tes\TESLocalAssemblerInner-impl.h	293	ProcessLib
warning C4244: 'argument' : conversion from 'const __int64' to 'unsigned int', possible loss of data (C:\Projects\ogs\ProcessLib\TES\TESLocalAssemblerInner.cpp)	c:\projects\ogs\processlib\tes\TESLocalAssemblerInner-impl.h	304	ProcessLib
warning C4244: 'argument' : conversion from 'const __int64' to 'unsigned int', possible loss of data (C:\Projects\ogs\ProcessLib\TES\TESLocalAssemblerInner.cpp)	c:\projects\ogs\processlib\tes\TESLocalAssemblerInner-impl.h	329	ProcessLib
warning C4244: 'argument' : conversion from 'const __int64' to 'unsigned int', possible loss of data (C:\Projects\ogs\ProcessLib\TES\TESLocalAssemblerInner.cpp)	c:\projects\ogs\processlib\tes\TESLocalAssemblerInner-impl.h	331	ProcessLib
warning C4244: 'argument' : conversion from 'const __int64' to 'unsigned int', possible loss of data (C:\Projects\ogs\ProcessLib\TES\TESLocalAssemblerInner.cpp)	c:\projects\ogs\processlib\tes\TESLocalAssemblerInner-impl.h	335	ProcessLib
warning C4244: 'argument' : conversion from 'const __int64' to 'unsigned int', possible loss of data (C:\Projects\ogs\ProcessLib\TES\TESLocalAssemblerInner.cpp)	c:\projects\ogs\processlib\tes\TESLocalAssemblerInner-impl.h	344	ProcessLib
warning C4244: 'argument' : conversion from 'const double' to 'unsigned __int64', possible loss of data	C:\Projects\ogs\NumLib\TimeStepping\Algorithms\FixedTimeStepping.cpp	84	NumLib
warning C4244: 'initializing' : conversion from '__int64' to 'int', possible loss of data (C:\Projects\ogs\MathLib\LinAlg\Eigen\EigenLinearSolver.cpp)	c:\users\rinkk\.conan\data\eigen3\3.2.8\bilke\stable\package\5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9\eigen\src\SparseLU\SparseLU.h	733	MathLib
warning C4244: 'initializing' : conversion from '__int64' to 'int', possible loss of data (C:\Projects\ogs\MathLib\LinAlg\Eigen\EigenLinearSolver.cpp)	c:\users\rinkk\.conan\data\eigen3\3.2.8\bilke\stable\package\5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9\eigen\src\SparseLU\SparseLU.h	734	MathLib
warning C4244: 'initializing' : conversion from '__int64' to 'int', possible loss of data (C:\Projects\ogs\MathLib\LinAlg\Eigen\EigenLinearSolver.cpp)	c:\users\rinkk\.conan\data\eigen3\3.2.8\bilke\stable\package\5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9\eigen\src\SparseLU\SparseLU_SupernodalMatrix.h	236	MathLib
warning C4244: 'initializing' : conversion from '__int64' to 'int', possible loss of data (C:\Projects\ogs\MathLib\LinAlg\Eigen\EigenLinearSolver.cpp)	c:\users\rinkk\.conan\data\eigen3\3.2.8\bilke\stable\package\5ab84d6acfe1f23c4fae0ab88f26e3a396351ac9\eigen\src\SparseLU\SparseLU_SupernodalMatrix.h	237	MathLib
warning C4800: 'int' : forcing value to bool 'true' or 'false' (performance warning) (C:\Projects\ogs\GeoLib\PointVec.cpp)	C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\include\utility	144	GeoLib
warning C4800: 'int' : forcing value to bool 'true' or 'false' (performance warning) (C:\Projects\ogs\GeoLib\Surface.cpp)	C:\Program Files (x86)\Microsoft Visual Studio 12.0\VC\include\utility	144	GeoLib
```

--- a/Applications/FileIO/FEFLOW/FEFLOWMeshInterface.cpp
+++ b/Applications/FileIO/FEFLOW/FEFLOWMeshInterface.cpp
@@ -380,7 +380,7 @@ void FEFLOWMeshInterface::readElevation(std::ifstream& in,
     std::size_t l = 0;
     unsigned mode = 0;  // 0: exit, 1: slice no, 2: elevation value, 3:
                         // continued line of mode 2
-    int pos_prev_line = 0;
+    std::streamoff pos_prev_line = 0;
     while (true)
     {
         pos_prev_line = in.tellg();

--- a/Applications/FileIO/FEFLOW/FEFLOWMeshInterface.cpp
+++ b/Applications/FileIO/FEFLOW/FEFLOWMeshInterface.cpp
@@ -482,7 +482,7 @@ MeshLib::Element* FEFLOWMeshInterface::readElement(
     switch (elem_type)
     {
         default:
-            for (unsigned k(0); k < n_nodes_of_element; ++k)
+            for (std::size_t k(0); k < n_nodes_of_element; ++k)
                 ele_nodes[k] = nodes[idx[k] - 1];
             break;
         case MeshLib::MeshElemType::HEXAHEDRON:

--- a/Applications/FileIO/Gmsh/GMSHAdaptiveMeshDensity.cpp
+++ b/Applications/FileIO/Gmsh/GMSHAdaptiveMeshDensity.cpp
@@ -115,7 +115,8 @@ void GMSHAdaptiveMeshDensity::getSteinerPoints (std::vector<GeoLib::Point*> & pn
             if ((*it)->getDepth() + additional_levels > max_depth) {
                 additional_levels = max_depth - (*it)->getDepth();
             }
-            const std::size_t n_pnts_per_quad_dim = 1 << additional_levels;
+            const std::size_t n_pnts_per_quad_dim = static_cast<std::size_t>(1)
+                                                    << additional_levels;
             const double delta ((ur[0] - ll[0]) / (2 * n_pnts_per_quad_dim));
             for (std::size_t i(0); i<n_pnts_per_quad_dim; i++) {
                 for (std::size_t j(0); j<n_pnts_per_quad_dim; j++) {

--- a/NumLib/TimeStepping/Algorithms/FixedTimeStepping.cpp
+++ b/NumLib/TimeStepping/Algorithms/FixedTimeStepping.cpp
@@ -80,7 +80,8 @@ FixedTimeStepping::newInstance(BaseLib::ConfigTree const& config)
     // append last delta_t until t_end is reached
     if (t_curr <= t_end)
     {
-        auto const repeat = std::ceil((t_end - t_curr) / delta_t);
+        auto const repeat =
+            static_cast<std::size_t>(std::ceil((t_end - t_curr) / delta_t));
         timesteps.resize(timesteps.size() + repeat, delta_t);
     }
 

--- a/ProcessLib/TES/TESLocalAssemblerInner-impl.h
+++ b/ProcessLib/TES/TESLocalAssemblerInner-impl.h
@@ -226,8 +226,10 @@ void TESLocalAssemblerInner<Traits>::assembleIntegrationPoint(
 {
     preEachAssembleIntegrationPoint(integration_point, localX, sm);
 
-    auto const N = sm.dNdx.cols();  // number of integration points
-    auto const D = sm.dNdx.rows();  // global dimension: 1, 2 or 3
+    auto const N =
+        static_cast<unsigned>(sm.dNdx.cols());  // number of integration points
+    auto const D =
+        static_cast<unsigned>(sm.dNdx.rows());  // global dimension: 1, 2 or 3
 
     assert(N * NODAL_DOF == local_M.cols());
 

--- a/Tests/NumLib/TestComponentNorms.cpp
+++ b/Tests/NumLib/TestComponentNorms.cpp
@@ -48,10 +48,10 @@ struct DOFTableData
     NumLib::LocalToGlobalIndexMap const dof_table;
 
 private:
-    static const std::size_t mesh_length;
+    static const double mesh_length;
     static const std::size_t mesh_elements_in_each_direction;
 };
-const std::size_t DOFTableData::mesh_length = 1;
+const double DOFTableData::mesh_length = 1;
 const std::size_t DOFTableData::mesh_elements_in_each_direction = 5;
 
 template <typename AccumulateCallback, typename AccumulateFinishCallback>

